backend-rust/src/graphql_api/baker.rs

```diff
@@ -378,26 +404,51 @@ enum BakerSort {
 struct BakerPool<'a> {
     /// Total stake of the baker pool as a percentage of all CCDs in existence.
     /// Includes both baker stake and delegated stake.
-    total_stake_percentage: Decimal,
+    total_stake_percentage:  Decimal,
     /// The total amount staked in this baker pool. Includes both baker stake
     /// and delegated stake.
-    total_stake:            Amount,
+    total_stake:             Amount,
     /// The total amount staked by delegators to this baker pool.
-    delegated_stake:        Amount,
+    delegated_stake:         Amount,
     /// The number of delegators that delegate to this baker pool.
-    delegator_count:        i64,
+    delegator_count:         i64,
+    /// The `commission_rates` represent the current commission settings of a
+    /// baker pool, as configured by the baker account through
+    /// `bakerConfiguration` transactions. These values are updated
+    /// immediatly by observing the following `BakerEvent`s in the indexer:
+    /// - `BakerSetBakingRewardCommission`
+    /// - `BakerSetTransactionFeeCommission`
+    /// - `BakerSetFinalizationRewardCommission`
+    ///  
+    /// Both `commission_rates` and `payday_commission_rates` are optional:  
+    /// - When a validator is initially added, only `commission_rates` are
+    ///   available until the next payday.
+    /// - When a validator is removed, we do not currently clean-up the
+    ///   database. Hence, the `payday_commission_rates`/`commission_rates` are
+    ///   the last observed value in that case.
+    commission_rates:        CommissionRates,
+    /// The `payday_commission_rates` represent the commission settings at the
+    /// last payday block. These values are retrieved from the
+    /// `get_bakers_reward_period(BlockIdentifier::AbsoluteHeight(payday_block_height))`  
+    /// endpoint at each payday.  
+    ///  
+    /// Both `commission_rates` and `payday_commission_rates` are optional:  
+    /// - When a validator is initially added, only `commission_rates` are
+    ///   available until the next payday.
+    /// - When a validator is removed, we do not currently clean-up the
+    ///   database. Hence, the `payday_commission_rates`/`commission_rates` are
+    ///   the last observed value in that case.
+    payday_commission_rates: CommissionRates,
```

Comment:
Since this is optional, I assume we need something like:
```suggestion
    payday_commission_rates: Option<CommissionRates>,
```
There would be no value until the first payday for a baker

Reply:
This case is currently represented as:
```
struct CommissionRates {
    transaction_commission:  Option<Decimal>,
    finalization_commission: Option<Decimal>,
    baking_commission:       Option<Decimal>,
}
```